pydeseq2/DeseqDataSet.py

```diff
@@ -176,11 +176,17 @@ def __init__(
         self.counts = counts
 
         # Import clinical data and convert design_column to string
-        self.clinical = clinical
-        if not self.counts.index.identical(self.clinical.index):
-            raise ValueError(
-                "The count matrix and clinical data should have the same index."
+        try:
+            self.clinical = clinical.loc[self.counts.index]
+        except KeyError as e:
+            print(
+                "The count matrix and clinical data "
+                "should contain the same sample indexes."
             )
+            raise e
+        assert len(self.clinical) == len(
```

Comment:
Why do you raise an assertion error here and not a KeyError again ? Unless I misunderstood the message is the same no ?

Reply:
I don't know, maybe I could raise a KeyError again... The reason behind the assertion is that `clinical.loc[self.counts.index]` will work when the index of `counts` is a subset of the index of `clinical`, but will drop rows from `clinical`.

Reply:
Simplified this part in [2d7ce6e](https://github.com/owkin/PyDESeq2/pull/30/commits/2d7ce6eb4e3246a12ebab69a2a9a3c0b95c147fc)